content/chainguard/chainguard-images/videos/migrating_go.md

```diff
@@ -0,0 +1,232 @@
+
+I should be able to access it.
+
+Okay, so that all works.
+
+It's a really nice, simple Go web server.
+
+Let's take a look at how we can change it to use a Chainguard Image.
+
+The easiest thing we can do, literally a one-line change, is just to modify this to point out the Chainguard Go image.
+
+With any luck, that will build just the same.
+
+So I'll go back to this original build statement.
+
+We'll call it something different so we can compare it later.
+
+Okay, that's built.
+
+Now we want to check it runs.
+
+I do need to first stop the old one.
+
+Okay, let's try running it again.
+
+It's running.
+
+Let's check it works.
+
+Okay, so that works identically.
+
+A one-line change, and we've made no difference to the actual running application.
+
+But let's investigate this a little bit more.
+
+So if I do 'docker images' on go-web-app, well this is a pretty large image at 892 megabytes.
+
+If I run it on our new image, it's 775 megabytes.
+
+So that's 120 megabytes saving, but still quite a large image.
+
+More interestingly, if I try Docker Scout Quick View to take a look at the CVEs, if I run it on Go web app, we can see there's 42 lower vulnerabilities in there.
+
+So there's quite a bit going on in there.
+
+We can also see there's 303 packages.
+
+So there's a lot of stuff in this image.
+
+Let's compare it to the Chainguard Image.
+
+There's no CVEs and there's only 85 packages.
+
+So this one change has really improved the security posture by getting rid of these 42 low CVEs.
+
+But there's a lot more we can do.
+
+Let's take a look at the Dockerfile again.
+
+So we're using this Chainguard Go image, but we don't need everything in this image when we're actually running the image.
+
+We just need the Go compiler, for instance, to build the server, but we don't need the Go compiler in the final image.
+
+So what we can do is create a multi-stage build.
+
+And I'm going to use the Chainguard static image to house the final production application.
+
+So the static image is really simple.
+
+It has very little in it, just sort of the minimum you need to run a typical Linux application.
+
+So it has TLS certificates so you can talk to web applications over TLS.
+
+It has Unix directories like /tmp and /home, and it doesn't have a lot else.
+
+It's only a few megabytes in size, but contains a few more things than say scratch that you need for typical applications.
+
+OK, what we're going to do here is I'm going to copy from the previous build to the Dockerfile up here.
+
+I'm going to copy the hello executable, which is at /work hello to /hello.
+
+Now it said copy from builder, so I'll need to name this first step builder.
+
+And that will build this hello executable and copy it into my production image.
+
+And then my entry point is now /hello.
+
+So there is one more thing we need to do.
+
+The static image does not include any libraries like glibc or anything.
+
+It just has it's just for running statically compiled binaries.
```

Comment:
I think something is missing in this line

Reply:
Should be fixed.